### ISSUE TITLE
Command Timeout - status failed

### ISSUE THREAD

**Issue by Smexhy** [AnalogJ/scrutiny#260]
SMART attribute Command Timeout - ID 188 (0xBC) is showing as failed on my 7 months old 5TB HDD. I rechecked in every other SMART program like CrystalDiskInfo and HardDiskSentinel but there wasn't any issue with that specific attribute (value was at 100, raw value is at 2). Did many surface tests and well and there wasn't any error mentioned. It kind of bugs me that it's showing as failed so I tried to dig around, reconnect SATA cable etc. multiple times, but the error was still there on scrutiny. Any idea how would I be able to narrow down what's wrong or how to perhaps make scrutiny not mark the disk as "failed" because of this?

**Comment by shamoon** [comment]
Yea I think a lot of folks are going to be asking about this, see https://github.com/AnalogJ/scrutiny/issues/247 https://github.com/AnalogJ/scrutiny/issues/187#issuecomment-1125355320 https://github.com/AnalogJ/scrutiny/issues/20 (etc) especially his comment here: https://github.com/AnalogJ/scrutiny/issues/247#issuecomment-1134859942

**Comment by Smexhy** [comment]
> Yea I think a lot of folks are going to be asking about this, see #247 [#187 (comment)](https://github.com/AnalogJ/scrutiny/issues/187#issuecomment-1125355320) #20 (etc) especially his comment here: [#247 (comment)](https://github.com/AnalogJ/scrutiny/issues/247#issuecomment-1134859942)

Yeah I thought this wouldn't be a real "failed" status as I checked it multiple times elsewhere and there wasn't any issue that I could find. Did you manage to fix it somehow or are we waiting for an update?

**Comment by AnalogJ** [comment]
I'm working on an update to disable NVMe SCRUTINY thresholds right now.

**Comment by shamoon** [comment]
I think author is aware of it =)

**Comment by Smexhy** [comment]
> I'm working on an update to disable NVMe SCRUTINY thresholds right now.

This one is however normal SCSI 2.5" HDD, will the update be containing something related to that as well?

**Comment by AnalogJ** [comment]
@Smexhy 
In your case, Scrutiny is working as intended. 

There's 2 different mechanisms that Scrutiny uses to detect failures.

The first is simple SMART failures. If SMART thinks an attribute is in a failed state, Scrutiny will display it as failed as well.

The second is using BackBlaze failure data: [backblaze.com/blog-smart-stats-2014-8.html](https://www.backblaze.com/blog-smart-stats-2014-8.html) If Scrutiny detects that an attribute corresponds with a high rate of failure using BackBlaze's data, it will also mark that attribute (and disk) as failed.

For your **"Command Timeout"** attribute, this is the exact data that Scrutiny is looking at:
https://www.backblaze.com/blog-smart-stats-2014-8.html#S188R

Basically if your disk has a value greater than `13G` as the raw value, that corresponds with an annual failure rate greater than 20%, so Scrutiny is telling you that your drive is likely to fail. 


The NVMe and SCSI changes that I am making are due to false-positives, with inaccurate data/thresholds. In your case this data is accurate.

I'll close this issue for now, but feel free to reopen/comment if you have any other questions!

**Comment by daveln** [comment]
@AnalogJ 
I may be mistaken here. But if I read the graph on https://www.backblaze.com/blog-smart-stats-2014-8.html#S188R for Command Timeout, I see that for a raw value from 0 to 13,000,000,000 there is a failure rate of 0.1 [0.081 - 0.123] and for a raw value from 13,000,000,000 to 26,000,000,000 there is a failure rate of 0.344 [0.252 - 0.434].

But the threshold data for Command Timeout in ata_attribute_metadata.go indicates an AnnualFailureRate of 0.10044174089362015 for a value from 0 to 13, and an AnnualFailureRate 0.334030592234279 for a value of 13 to 26.

Shouldn't the Low and High threshold data for Command Timeout in ata_attribute_metadata.go be multiplied by 1,000,000,000? Is there another source where these thresholds come from?

**Comment by Smexhy** [comment]
> @AnalogJ I may be mistaken here. But if I read the graph on https://www.backblaze.com/blog-smart-stats-2014-8.html#S188R for Command Timeout, I see that for a raw value from 0 to 13,000,000,000 there is a failure rate of 0.1 [0.081 - 0.123] and for a raw value from 13,000,000,000 to 26,000,000,000 there is a failure rate of 0.344 [0.252 - 0.434].
> 
> But the threshold data for Command Timeout in ata_attribute_metadata.go indicates an AnnualFailureRate of 0.10044174089362015 for a value from 0 to 13, and an AnnualFailureRate 0.334030592234279 for a value of 13 to 26.
> 
> Shouldn't the Low and High threshold data for Command Timeout in ata_attribute_metadata.go be multiplied by 1,000,000,000? Is there another source where these thresholds come from?

This is very interesting finding to be honest. @AnalogJ I wanted to ask however if it is possible for me as a user to skip certain checks for specific parameters, in this case for the Command Timeout. Would I be somehow able to set in scrutiny threshold for this value when it wouldn't show the disk as "failed"? I had this value for over a year now and the disk works just fine in this case. I would love to just ignore it either completely for this specific attribute and disk or just set the threshold for myself, so it would show as "passed". If it's currently not possible, I think it would be decent feature request for some special cases.

**Comment by vasishche** [comment]
> Basically if your disk has a value greater than `13G` as the raw value, that corresponds with an annual failure rate greater than 20%, so Scrutiny is telling you that your drive is likely to fail.
> 
> I'll close this issue for now, but feel free to reopen/comment if you have any other questions!

@AnalogJ I have a value much lower than 13G, but it shows a failure.

<img width="1002" height="259" alt="Image" src="https://github.com/user-attachments/assets/149a1a0b-e7ff-41b5-b921-da8c64fe6169" />

**Comment by DanaGoyette** [comment]
Same for two different Micron Ion 5210 SATA SSDs:

<img width="2024" height="1340" alt="Image" src="https://github.com/user-attachments/assets/3c6dfbe8-38a5-42f8-bf91-0aa96dca2a70" />

<img width="2006" height="1330" alt="Image" src="https://github.com/user-attachments/assets/57971cf4-818a-47ed-b0fd-ad393e765494" />
